Source/Kernel/Grains/Observation/Jobs/HandleEventsForPartition.cs

```diff
@@ -21,108 +22,122 @@ namespace Cratis.Chronicle.Grains.Observation.Jobs;
 /// </remarks>
 /// <param name="state"><see cref="IPersistentState{TState}"/> for managing state of the job step.</param>
 /// <param name="storage"><see cref="IStorage"/> for accessing storage for the cluster.</param>
+/// <param name="logger">The logger.</param>
 public class HandleEventsForPartition(
     [PersistentState(nameof(JobStepState), WellKnownGrainStorageProviders.JobSteps)]
     IPersistentState<HandleEventsForPartitionState> state,
-    IStorage storage) : JobStep<HandleEventsForPartitionArguments, HandleEventsForPartitionResult, HandleEventsForPartitionState>(state), IHandleEventsForPartition
+    IStorage storage,
+    ILogger<HandleEventsForPartition> logger) : JobStep<HandleEventsForPartitionArguments, HandleEventsForPartitionResult, HandleEventsForPartitionState>(state), IHandleEventsForPartition
 {
     const string SubscriberDisconnected = "Subscriber is disconnected";
 
     IEventSequenceStorage? _eventSequenceStorage;
-    IObserver? _observer;
+    IObserver _observer = null!;
+    EventSourceId _eventSourceId = EventSourceId.Unspecified;
     IObserverSubscriber? _subscriber;
 
     /// <inheritdoc/>
     public override Task Prepare(HandleEventsForPartitionArguments request)
     {
-        var eventSourceId = (EventSourceId)(request.Partition.Value.ToString() ?? string.Empty);
+        using var scope = logger.BeginObserverScope(request.ObserverKey, State.Id.JobId, State.Id.JobStepId);
+        logger.Preparing(request.Partition, request.StartEventSequenceNumber, request.EndEventSequenceNumber);
         _observer = GrainFactory.GetGrain<IObserver>(request.ObserverKey);
+        _eventSourceId = request.Partition.Value?.ToString() ?? EventSourceId.Unspecified;
```

Comment:
`Key` has an override for `.ToString()`. The `Value` within `Key` is not nullable, so I think this can be simplified to:

```suggestion
        _eventSourceId = request.Partition.ToString();
```